drift_dev/lib/src/model/types.dart

```diff
@@ -75,7 +75,7 @@ class DriftDartType {
 extension OperationOnTypes on HasType {
   /// Whether this type is nullable in Dart
   bool get nullableInDart {
-    return (nullable && !isArray) || typeConverter?.hasNullableDartType == true;
```

Comment:
This may have to change in the future, but I believe your changes are correct now.

Once we get nullability with type converters right in the next breaking version, there could be a way to apply non-nullable type converters onto a nullable column, in which case the old method is correct because the result would still be nullable.

Reply:
It can be nullable. But this is the responsibility of user. If column is not nullable, converter can still return null in any case. It was true before this dartType depend nullability.

Reply:
I think this should be something the following now:

```dart
if (isArray) return false; // Returns a non-nullable `List`

final converter = typeConverter;
if (converter != null) {
  // Nullable if the type converter is nullable in Dart or if nulls side-step the converter.
  return converter.hasNullableDartType || (converter.canBeSkippedForNulls && nullable);
}
return nullable;
```